### ISSUE TITLE
Auto-detect conflicts/overlaps in rebuild TODO lists

### ISSUE THREAD

**Issue by lfos** [archlinux/archweb#205]
One thing that has bothered me a couple of times is that archweb does not automatically detect potential conflicts/overlaps between different rebuilds even if it could. This sounds like a feature that is easy to implement and quite useful at the same time.

First draft of how it could work:

* When viewing a TODO list of an incomplete TODO, at the top (below the description?), display a warning that the package list has a non-empty intersection with another incomplete TODO list.
* Additionally, either provide the list of overlapping packages next to that warning or add a separate column to the package list that indicates overlaps.
* Maybe also add an option to exclude a TODO list from the overlap detection (to be used for non-conflicting TODOs like the HTTPs sources rebuilds) such that we do not get false positives. Alternatively, add different types of TODO lists for rebuilds, source-only fixes --- not sure whether that is a great idea, though.

**Comment by jelly** [comment]
Thanks for creating the issue, this has come up before. But I was thinking about hinting about overlapping packages when creating a todo list which seems to have come up before if I recall correctly.

A different idea about implementing the UX for this feature would be to add either a column to the todolist of a "red" background color if a package exists in another todolist.

**Comment by lfos** [comment]
> A different idea about implementing the UX for this feature would be to add either a column to the todolist of a "red" background color if a package exists in another todolist.

Agreed. But there should be a way to identify the conflicting TODO list easily. A "careful, you need to coordinate this rebuild with something" is certainly much less useful than "careful, you need to coordinate this rebuild with `$pkg`".

> I'm not entirely sure about todolist types, we could introduce a checkbox "check overlap todolist?" or create todolist types such as "general gripe" or "rebuilds". Since I guess the issue is really with rebuilds?

That's actually what I was thinking of when suggesting "TODO list types" :)

**Comment by jelly** [comment]
The first step has finally been taken, by adding a todolist type ce24198793d59ad0d6d6f8277fb4eb90ea05cdd7 to archweb. Next up is adding overlapping packages to the todolist view.
